test/xpu_api/tuple/tuple.tuple/tuple.creation/tuple_cat.pass.cpp

```diff
@@ -0,0 +1,289 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/tuple>
+#include <oneapi/dpl/utility>
+#include <oneapi/dpl/array>
+
+#include "support/utils.h"
+#include "support/utils_invoke.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+struct MoveOnlyData
+{
+    int idx_ = 0;
+
+    MoveOnlyData(int idx) : idx_(idx) {}
+    MoveOnlyData(MoveOnlyData&& x) = default;
+    MoveOnlyData(const MoveOnlyData&) = delete;
+
+    MoveOnlyData& operator=(MoveOnlyData&& x) = default;
+    MoveOnlyData& operator=(const MoveOnlyData&) = delete;
+
+    bool
+    operator==(const MoveOnlyData& x) const
+    {
+        return idx_ == x.idx_;
+    }
+    bool
+    operator<(const MoveOnlyData& x) const
+    {
+        return idx_ < x.idx_;
+    }
+    MoveOnlyData
+    operator+(const MoveOnlyData& x) const
+    {
+        return MoveOnlyData{idx_ + x.idx_};
+    }
+    MoveOnlyData
+    operator*(const MoveOnlyData& x) const
+    {
+        return MoveOnlyData{idx_ * x.idx_};
+    }
```

Comment:
I have looked through the test and it seems like these functions are not necessary

Reply:
Fixed